scripts/update-appstream.mjs

```diff
@@ -0,0 +1,44 @@
+import fs from 'node:fs/promises';
```

Comment:
pushing to releases will soon make it quite big(including beta and canary). if the old releases aren't that important, we could simply replace the `version=x` to the latest.

Reply:
Oh, set-version is being executed for beta and canary as well? I'd exclude those. For stable releases I don't think that's too big of a concern. And even if it becomes one - you could migrate to external releases file: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases

I double-checked Flathub submission guidelines, they don't explicitly state that all releases need to be included. But I'd prefer to keep it in, it would probably affect how app store render version history.

Reply:
if a downgrade option is not provided then the version history is meaningless, and we can't promise downgrading the app won't cause data lost cause it's hard to do forward-compatibility. I will directly update your pr before it gets merged.